src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -179,7 +179,9 @@ parseTypeClassDeclaration = do
   return $ TypeClassDeclaration className idents implies members
 
 parseConstraint :: TokenParser Constraint
-parseConstraint = (,) <$> parseQualified properName <*> P.many (noWildcards parseTypeAtom)
+parseConstraint = Constraint <$> parseQualified properName
```

Comment:
I'm a bit worried that this didn't cause the tests to fail, actually, unless this instance just happens to match exactly up with the derived `ToJSON` instance in `Language.PureScript.Types`.